tooling/src/main/java/org/opencds/cqf/tooling/casereporting/tes/TESPackageGenerator.java

```diff
@@ -0,0 +1,615 @@
+            List<Coding> reportingSpecificationGrouperCodes = new ArrayList<>();
+
+            for (ValueSet reportingSpecificationGrouper : reportingSpecificationGroupers) {
+                Optional<UsageContext> maybeUseContext =
+                    reportingSpecificationGrouper.getUseContext().stream().filter(uc -> uc.getCode().getCode().equalsIgnoreCase("focus")).findFirst();
+
+                if (maybeUseContext.isPresent()) {
+                    var useContext = maybeUseContext.get().getValueCodeableConcept().getCodingFirstRep();
+                    reportingSpecificationGrouperCodes.add(new Coding(useContext.getSystem(), useContext.getCode(), maybeUseContext.get().getValueCodeableConcept().getText()));
+                }
+            }
+
+            // Create a new workbook and sheet
+            String filePath = getOutputPath() + "/condition-code-diff.xlsx";
+            Workbook workbook = new XSSFWorkbook(); //WorkbookFactory.create(conditionCodeDiffWorkbookFile);
+
+            // Generate a Sheet with list of codes that are in the RCKMS Condition Code ValueSet,
+            // but not implemented in any current reporting specifications
+            Sheet sheet1 = workbook.createSheet("In VS, not in Production");
+            writeDifferencesToSheet(conditionCodeValueSetCodes, reportingSpecificationGrouperCodes, sheet1);
+
+            // Generate a Sheet with list of codes that are associated with current reporting
+            // specification implementations, but not in the RCKMS Condition Code ValueSet.
+            Sheet sheet2 = workbook.createSheet("In Production, not in VS");
+            writeDifferencesToSheet(reportingSpecificationGrouperCodes, conditionCodeValueSetCodes, sheet2);
+
+            // Save the workbook to the file system
+            try (FileOutputStream fileOut = new FileOutputStream(filePath)) {
+                workbook.write(fileOut);
+                System.out.println("Excel file written successfully.");
+            } catch (IOException e) {
+                System.out.println("Error writing Excel file: " + e.getMessage());
+            } finally {
+                try {
+                    workbook.close();
+                } catch (IOException e) {
+                    System.out.println("Error closing workbook: " + e.getMessage());
+                }
+            }
+        }
+    }
+
+    private void writeDifferencesToSheet(List<Coding> baseList, List<Coding> comparisonList, Sheet sheet) {
+        int rowCounter = 0;
+        Row headerRow = sheet.createRow(rowCounter);
+        Cell headerSystemCell = headerRow.createCell(0);
+        headerSystemCell.setCellValue("System");
+        Cell headerCodeCell = headerRow.createCell(1);
+        headerCodeCell.setCellValue("Code");
+        Cell headerDisplayCell = headerRow.createCell(2);
+        headerDisplayCell.setCellValue("Text");
+        rowCounter++;
+
+        for (int i = 0; i < baseList.size(); i++) {
```

Comment:
Old school - not that there is anything wrong with that ;). Enhanced for-loops are typically preferred